fix(linter): ignore invalid or partial disable directives

camchenry opened the pull request:
- closes https://github.com/oxc-project/oxc/issues/6041

We were not ignoring cases where the directive was invalid, now we do.

camchenry commented:
* **#6045** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6045?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @camchenry and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

DonIsaac commented on crates/oxc_linter/src/disable_directives.rs:
```diff
@@ -161,17 +159,23 @@ impl<'a> DisableDirectivesBuilder<'a> {
+                // Like `eslint-disable-lext-nine` where `text` is `-lext-nine`
+                !text.starts_with('-') &&
+                    // We should match only `eslint-disable` comments, not `eslint-disablefoo`
+                    text.starts_with(' ')
```
This condition is logically identical to `text.starts_with(' ')`. You should be able to remove the `!text.starts_with('-')` part.

camchenry committed:
fix(linter): ignore invalid or partial disable directives (#6045)

- closes https://github.com/oxc-project/oxc/issues/6041

We were not ignoring cases where the directive was invalid, now we do.
